packages/core/src/embeddings/OpenAIEmbedding.ts

```diff
@@ -16,6 +16,8 @@ export enum OpenAIEmbeddingModelType {
 export class OpenAIEmbedding extends BaseEmbedding {
   model: OpenAIEmbeddingModelType | string;
 
+  embedBatchSize: number;
```

Comment:
doesn't use batching yet, needs to send one batch at once to openai api

Reply:
@marcusschiesser the batching will be used when updated on `VectorStoreIndex`, correct?

Reply:
Once `VectorStoreIndex` is updated to use batching, the batching method from `BaseEmbedding` will be used, yes.
But what I mean is that the `OpenAIEmbedding` implementation should use the batching support of the OpenAI Embedding API (see https://platform.openai.com/docs/api-reference/embeddings/create#embeddings-create-input):
The `input` parameter supports passing a `string[]` to generate multiple embeddings at once. 
Most of the other APIs have similar support, but we can add this step-by-step in separate PRs.

Reply:
@marcusschiesser 

Thanks for the detailed explanation

I'm updating it now, wasn't ware of the batch on OpenAI side as well

Reply:
@EDIT tbh honest I think currently works on this way, it also can be manipulated by the chunk size, maybe we can later further discuss more bout it

Reply:
To follow up on Marcus's comment here, what we're looking for in this feature is to probably use OpenAI/the underlying provdier's batching functionality.

Reply:
I can look into it, but the feature is to help avoid rate limiting, even if we fill the input[] with more text and send a lot of requests it will rate limit, so that's why the configurable embedBatchSize

Reply:
But for sure adding more text into the batch will make it faster and reduce the amounts of requests as well, so it's 2 parts of the feature

Reply:
@EmanuelCampos you're right, these are two concerns, and we can split them into two PRs.

One advantage of doing it now is that you can factor out your `processChunk` function below on class level and then override `processChunk` here in the OpenAIEmbedding to send multiple strings in one request (a few lines of code).

You won't even need then here the `embedBatchSize`